src/dom/react-nodegui/src/components/Svg/RNSvg.ts

```diff
@@ -0,0 +1,218 @@
+   * Sets whether mouse tracking is enabled for this widget
+   */
+  mouseTracking?: boolean;
+
+  /**
+   * Sets the parent node for this widget
+   */
+  nodeParent?: Component;
+
+  /**
+   * Sets the stylesheet
+   */
+  styleSheet?: string;
+
+  /**
+   * Sets the file path
+   */
+  src?: string | Buffer;
+}
+
+const setSvgProps = (
+  widget: RNSvg,
+  newProps: SvgProps,
+  oldProps: SvgProps
+) => {
+  const setter: SvgProps = {
+    set acceptDrops(on: boolean) {
+      widget.setAcceptDrops(on);
+    },
+    set cursor(cursor: CursorShape|QCursor) {
+      widget.setCursor(cursor);
+    },
+    set enabled(enabled: boolean) {
+      widget.setEnabled(enabled);
+    },
+    set fixedSize({width, height}: {width: number, height: number}) {
+      widget.setFixedSize(width, height);
+    },
+    set flexNodeSizeControlled(isSizeControlled: boolean) {
+      widget.setFlexNodeSizeControlled(isSizeControlled);
+    },
+    set font(font: QFont) {
+      widget.setFont(font);
+    },
+    set geometry({x, y, w, h}: {x: number, y: number, w: number, h: number}) {
+      widget.setGeometry(x, y, w, h);
+    },
+    set graphicsEffect(effect: QGraphicsEffect<any>) {
+      widget.setGraphicsEffect(effect);
+    },
+    set inlineStyle(style: string) {
+      widget.setInlineStyle(style);
+    },
+    set layout(parentLayout: NodeLayout<QWidgetSignals>) {
+      widget.setLayout(parentLayout);
+    },
+    set maximumSize({maxw, maxh}: {maxw: number, maxh: number}) {
+      widget.setMaximumSize(maxw, maxh);
+    },
+    set minimumSize({minw, minh}: {minw: number, minh: number}) {
+      widget.setMinimumSize(minw, minh);
+    },
+    set mouseTracking(isMouseTracked: boolean) {
+      widget.setMouseTracking(isMouseTracked);
+    },
+    set nodeParent(parent: Component) {
+      widget.setNodeParent(parent);
+    },
+    set styleSheet(styleSheet: string) {
+      widget.setStyleSheet(styleSheet);
+    },
+    set src(file: string | Buffer) {
+      widget.load(file);
+    },
+    set id(id: string) {
+      widget.setObjectName(id);
+    },
+    set on(
+      listenerMap: Partial<WidgetEventListeners | QWidgetSignals>
+    ) {
+      const listenerMapLatest: any = Object.assign({}, listenerMap);
+      const oldListenerMap = Object.assign({}, oldProps.on);
+      Object.entries(oldListenerMap).forEach(([eventType, oldEvtListener]) => {
+        const newEvtListener = listenerMapLatest[eventType];
+        if (oldEvtListener !== newEvtListener) {
+          widget.removeEventListener(eventType as any, oldEvtListener);
+        } else {
+          delete listenerMapLatest[eventType];
+        }
+      });
+
+      Object.entries(listenerMapLatest).forEach(
+        ([eventType, newEvtListener]) => {
+          widget.addEventListener(eventType as any, newEvtListener);
+        }
+      );
+    },
+    set visible(shouldShow: boolean) {
+      shouldShow ? widget.show() : widget.hide();
+    },
+  };
+  Object.assign(setter, newProps);
```

Comment:
See [RNBoxView.ts](/src/dom/react-nodegui/components/BoxView/RNBoxView.ts) again. Just add the line `setViewProps(widget, newProps, oldProps)` after the Object.assign call and you should be able to remove most, if not all, of that setter code. Looking at the [QSvgWidget](https://github.com/nodegui/nodegui/blob/master/src/lib/QtWidgets/QSvgWidget.ts) source code, it doesn't actually appear to have any props that can be set beyond the view props, so `setViewProps` should actually cover everything. You'd best still pass an empty object in as a placeholder for consistency with the other widgets, as we may want to add support for extra props in future.